flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -320,11 +324,25 @@ protected void initializeServices(Configuration configuration, PluginManager plu
             metricRegistry.startQueryService(metricQueryServiceRpcService, null);
 
             final String hostname = RpcUtils.getHostname(commonRpcService);
+            FileJobGraphRetriever fileJobGraphRetriever =
+                    FileJobGraphRetriever.createFrom(configuration, null);
+
+            String jobName = null;
+            String jobId = null;
+            try {
+                JobGraph jobGraph = fileJobGraphRetriever.retrieveJobGraph(configuration);
+                jobName = jobGraph.getName();
+                jobId = jobGraph.getJobID().toHexString();
+            } catch (FlinkException e) {
+                LOG.info("Not able to fetch JobGraph");
+            }
```

Comment:
I think this only works for the Yarn per job cluster entrypoint and is not a general solution. For example, the application mode won't be supported. Hence, I think we cannot do it like this.